docs/ecosystem-user-event.mdx

```diff
@@ -204,6 +206,36 @@ test('delete characters within the selectedRange', () => {
   userEvent.type(input, '{backspace}good')
 
   expect(input).toHaveValue('This is a good example')
+```
+
+By default, `type` appends to the existing text. To prepend text, reset the
+element's selection range and provide the `initialSelectionStart` and
+`initialSelectionEnd` options:
+
+```jsx
+import React from 'react'
+import {render, screen} from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+
+test('prepend text', () => {
+  render(
+    <div>
+      <label htmlFor="my-input">Example:</label>
+      <input id="my-input" type="text" />
+    </div>,
+  )
+
+  const input = screen.getByLabelText(/example/i)
+  userEvent.type(element, 'World!')
+
+  // Prepend text
+  element.setSelectionRange(0, 0)
```

Comment:
```suggestion
```
The selection range is already at `0, 0`.

Reply:
Can we double-check this against user-event@13.5.0? It looks like the documentation is reorganized for user-event@14, but this page still targets version 13.

Running the following test against user-event@13.5.0 (https://github.com/testing-library/user-event/commit/5d946d51d643f0ef7e7730fa527b7ca96e330907) fails without the `element.setSelectionRange(0, 0)` line, because the selection range is `6, 6`:
```js
import userEvent from '../'
import React from 'react'
import {render, screen} from '@testing-library/react'

test('selection range', () => {
  render(<input defaultValue="World!"/>)
  const element = screen.getByRole('textbox')

  // Prepend text
  // element.setSelectionRange(0, 0)
  userEvent.type(element, 'Hello, ', {
    initialSelectionStart: 0,
    initialSelectionEnd: 0,
  })

  expect(element).toHaveValue('Hello, World!')
})
```

```
  ● selection range

    expect(element).toHaveValue(Hello, World!)

    Expected the element to have value:
      Hello, World!
    Received:
      World!Hello, 

      14 |   })
      15 |
    > 16 |   expect(element).toHaveValue('Hello, World!')
         |                   ^
      17 | })
      18 |

```

Reply:
You're right - React sets `selectionStart` to the end of the default value.